docs/specs/authentication-flow-api-reference.md

```diff
@@ -239,6 +266,134 @@ When you are in this step of this flow, you will see a response like the followi
 }
 ```
 
+### Risk assessment and Captcha
+
+Each option may contain the key `risk_assessment` and the key `captcha`.
+
+The shape of `risk_assessment`:
+
+```
+{
+  "required": true,
+  "provider": {
+    "type": "recaptchav3",
+    "alias": "recaptchav3"
+  }
+}
+```
+
+If `risk_assessment.required` is true, then selecting this option requires performing risk assessment first.
+
+You use `provider.type` to determine which client side library to use.
+
+You use `provider.alias` to determine which client side credentials to use.
+
+---
+
+The shape of `captcha`:
+
+```
+{
+  "required": true,
+  "provider": {
+    "type": "recaptchav3",
+    "alias": "recaptchav3"
+  }
+}
+```
+
+If `captcha.required` is true, then selecting this option requires performing captcha challenge first.
+
+You use `provider.type` to determine which client side library to use.
+
+You use `provider.alias` to determine which client side credentials to use.
+
+---
+
+If both are present, you must perform risk assessment first.
+
+#### Risk assessment input
+
+To perform risk assessment, pass an input of the following shape:
+
+```json
+{
+  "risk_assessment": {
+    "alias": "recaptchav3",
+    "type": "recaptchav3",
+    "response": { ... }
+  }
+}
+```
+
+- `risk_assessment.alias`: The alias of the risk assessment provider.
+- `risk_assessment.type`: The type of the risk assessment provider.
+
+Other fields are provider-specific.
+
+#### risk_assessment input; type: recaptchav3
+
+- `risk_assessment.response`: The response provided by the reCAPTCHA v3 client-side library.
```

Comment:
nitpick: How about we use `response_token` for this field?

> [response 	Required. The user response token provided by the reCAPTCHA client-side integration on your site.](https://developers.google.com/recaptcha/docs/verify#api_request)

Reply:
Do you mean renaming `response` to `response_token`? My original meaning is that `risk_assessment.response` is the `response` in `POST https://www.google.com/recaptcha/api/siteverify`.

Reply:
> Do you mean renaming response to response_token?

yes, but rather trivial, just wanna be more descriptive to user

> My original meaning is that risk_assessment.response is the response in POST https://www.google.com/recaptcha/api/siteverify.

I thought it's whatever the client-side library returns 😱